x/dsmr/node_test.go

```diff
@@ -1520,3 +1510,55 @@ func newTestNodes(t *testing.T, n int) []*Node[dsmrtest.Tx] {
 
 	return result
 }
+
+type testChainState struct {
+	validators []Validator
+	quorumNum  uint64
+	quorumDen  uint64
+}
+
+func (*testChainState) GetNetworkParams() (uint32, ids.ID, ids.ID) {
+	return networkID /*subnetID*/, ids.Empty, chainID
+}
+
+func (t *testChainState) getValidatorSet() map[ids.NodeID]*validators.GetValidatorOutput {
```

Comment:
I feel like this introduces complexity, can we just cache this in the `testChainState` struct instead of having to re-call this function every time you want to get the validator set

Reply:
The method is being used only in a single place ( within `GetCanonicalValidatorSet` ), and exclusively used for testing purposes. It doesn't seems that we would gain much by caching the map here. We could use a map as the backing storage ( instead of a slice ).